src/lightning/app/core/plugin.py

```diff
@@ -0,0 +1,167 @@
+    """Create a run with the given name and entrypoint under the cloudspace with the given ID."""
+    if run.app_id is None and run.plugin_name == "app":
+        from lightning.app.runners.cloud import CloudRuntime
+
+        # TODO: App dispatch should be a plugin
+        # Dispatch the run
+        _set_flow_context()
+
+        entrypoint_file = Path("/content") / run.entrypoint
+
+        app = CloudRuntime.load_app_from_file(str(entrypoint_file.resolve().absolute()))
+
+        app.stage = AppStage.BLOCKING
+
+        runtime = CloudRuntime(
+            app=app,
+            entrypoint=entrypoint_file,
+            start_server=True,
+            env_vars={},
+            secrets={},
+            run_app_comment_commands=True,
+        )
+        # Used to indicate Lightning has been dispatched
+        os.environ["LIGHTNING_DISPATCHED"] = "1"
+
+        try:
+            runtime.cloudspace_dispatch(
+                project_id=run.project_id,
+                cloudspace_id=run.cloudspace_id,
+                name=run.name,
+                cluster_id=run.cluster_id,
+            )
+        except Exception as e:
+            raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR, detail=str(e))
+    elif run.app_id is not None:
+        from lightning.app.utilities.cli_helpers import _LightningAppOpenAPIRetriever
+        from lightning.app.utilities.commands.base import _download_command
+
+        retriever = _LightningAppOpenAPIRetriever(run.app_id)
+
+        metadata = retriever.api_commands[run.plugin_name]  # type: ignore
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+
+            target_file = os.path.join(tmpdir, f"{run.plugin_name}.py")
+            plugin = _download_command(
+                run.plugin_name,
+                metadata["cls_path"],
+                metadata["cls_name"],
+                run.app_id,
+                target_file=target_file,
+            )
+
+            if isinstance(plugin, Plugin):
+                plugin._setup(app_id=run.app_id)
+                plugin.run(run.name, run.entrypoint)
+            else:
+                # This should never be possible but we check just in case
+                raise HTTPException(
+                    status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                    detail=f"The plugin {run.plugin_name} is an incorrect type.",
+                )
+    else:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST, detail="App ID must be specified unless `plugin_name='app'`."
+        )
+
+
+def _start_plugin_server(host, port) -> None:
+    """Start the plugin server which can be used to dispatch apps or run plugins."""
+    fastapi_service = FastAPI()
+
+    fastapi_service.add_middleware(
+        CORSMiddleware,
+        allow_origins=["*"],
+        allow_credentials=True,
+        allow_methods=["*"],
+        allow_headers=["*"],
+    )
+
+    fastapi_service.post("/v1/runs")(_run_plugin)
```

Comment:
You can dynamically add routes to fastapi. Once `_run_plugin` has been triggered, you can use the Input payload from the created Run and register new routes for the CloudSpace to interact with and trigger the user  run_app_command

Reply:
Going to leave this for a follow-up whilst we build out the plugin service - once we have some example plugins it will be easier to see how it can be improved